partitioned-heat-conduction/fenicsx/heat.py

```diff
@@ -0,0 +1,254 @@
+V_g = FunctionSpace(mesh, vector_element)
+W = V_g.sub(0).collapse()[0]
+
+# Define boundary conditions
+
+
+class Expression_u_D:
+    def __init__(self):
+        self.t = 0.0
+
+    def eval(self, x):
+        return np.full(x.shape[1], 1 + x[0] * x[0] + alpha * x[1] * x[1] + beta * self.t)
+
+
+u_D = Expression_u_D()
+u_D_function = Function(V)
+u_D_function.interpolate(u_D.eval)
+
+if problem is ProblemType.DIRICHLET:
+    # Define flux in x direction
+    class Expression_f_N:
+        def __init__(self):
+            pass
+
+        def eval(self, x):
+            return np.full(x.shape[1], 2 * x[0])
+
+    f_N = Expression_f_N()
+    f_N_function = Function(V)
+    f_N_function.interpolate(f_N.eval)
+
+# Define initial value
+u_n = Function(V, name="Temperature")
+u_n.interpolate(u_D.eval)
+
+precice, precice_dt, initial_data = None, 0.0, None
+
+# Initialize the adapter according to the specific participant
+if problem is ProblemType.DIRICHLET:
+    precice = Adapter(MPI.COMM_WORLD, adapter_config_filename="precice-adapter-config-D.json")
+    precice_dt = precice.initialize(coupling_boundary, read_function_space=V, write_object=f_N_function)
+elif problem is ProblemType.NEUMANN:
+    precice = Adapter(MPI.COMM_WORLD, adapter_config_filename="precice-adapter-config-N.json")
+    precice_dt = precice.initialize(coupling_boundary, read_function_space=W, write_object=u_D_function)
+
+dt = Constant(mesh, 0.0)
+dt.value = np.min([fenics_dt, precice_dt])
+
+# Define variational problem
+u = TrialFunction(V)
+v = TestFunction(V)
+
+
+class Expression_f:
+    def __init__(self):
+        self.t = 0.0
+
+    def eval(self, x):
+        return np.full(x.shape[1], beta - 2 - 2 * alpha)
+
+
+f = Expression_f()
+f_function = Function(V)
+F = u * v / dt * dx + dot(grad(u), grad(v)) * dx - (u_n / dt + f_function) * v * dx
+
+dofs_remaining = locate_dofs_geometrical(V, remaining_boundary)
+bcs = [dirichletbc(u_D_function, dofs_remaining)]
+
+coupling_expression = precice.create_coupling_expression()
+read_data = precice.read_data()
+precice.update_coupling_expression(coupling_expression, read_data)
+if problem is ProblemType.DIRICHLET:
+    # modify Dirichlet boundary condition on coupling interface
+    dofs_coupling = locate_dofs_geometrical(V, coupling_boundary)
+    bcs.append(dirichletbc(coupling_expression, dofs_coupling))
+if problem is ProblemType.NEUMANN:
+    # modify Neumann boundary condition on coupling interface, modify weak
+    # form correspondingly
+    F += v * coupling_expression * ds
+
+a, L = lhs(F), rhs(F)
+
+# Time-stepping
+u_np1 = Function(V, name="Temperature")
+t = 0
+
+# reference solution at t=0
+u_ref = Function(V, name="reference")
+u_ref.interpolate(u_D_function)
+
+
+xdmf = XDMFFile(MPI.COMM_WORLD, f"./out/{precice.get_participant_name()}.xdmf", "w")
+xdmf.write_mesh(mesh)
+ref_out = XDMFFile(MPI.COMM_WORLD, f"./out/ref{precice.get_participant_name()}.xdmf", "w")
+ref_out.write_mesh(mesh)
```

Comment:
```suggestion
xdmf = XDMFFile(MPI.COMM_WORLD, f"./output/{precice.get_participant_name()}.xdmf", "w")
xdmf.write_mesh(mesh)
ref_out = XDMFFile(MPI.COMM_WORLD, f"./output/ref{precice.get_participant_name()}.xdmf", "w")
ref_out.write_mesh(mesh)
```
Because the cleaning tool looks for the `output` folder.

Reply:
Changed in https://github.com/precice/tutorials/pull/317/commits/e0a3f4c425f98ce25f00a52b868a83b19aca038a.